### PR TITLE
adding code for changes in retry button

### LINKED ISSUE #2255: Retry button too close to scrollbar
2.9.0 recent alphas

Tapping the retry button is very difficult as it is dangerously close to the "X" button, and most taps when I try are taken by the scrollbar.

### PR DESCRIPTION
**Pull request by sashrikakaur**
Signed-off-by: Sashrika Kaur <sashrikakaur@yahoo.co.in>

The retry button was close to "X" 

Fixes #2255  Retry button too close to the scrollbar

What changes did you make and why?
Changed the position of the button to provide better UI in case the upload failed. Added space between the retry button and cancel upload button.

Tested betaDebug on MI A2 with API level 28

image

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
 #Thu Mar 01 15:28:48 IST 2018
-org.gradle.jvmargs=-Xmx1536M
+org.gradle.jvmargs=-Xmx2048M
```
Comment (maskaravivek) [inline gradle.properties]: Please undo this.

File: app/build.gradle [hunk 2]
```diff
@@ -95,6 +95,8 @@ dependencies {
     implementation 'com.daimajia.swipelayout:library:1.2.0@aar'
     //metadata extractor
     implementation 'com.drewnoakes:metadata-extractor:2.11.0'
+
+    compile group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.1'
```
Comment (maskaravivek) [inline app/build.gradle]: why is this needed?

File: build.gradle [hunk 3]
```diff
@@ -22,5 +22,6 @@ allprojects {
         jcenter()
         maven { url "https://jitpack.io" }
         maven { url "https://maven.google.com" }
+        mavenCentral()
```
Comment (maskaravivek) [inline build.gradle]: why is this needed?

File: app/src/main/res/layout/layout_contribution.xml [hunk 4]
```diff
@@ -92,16 +113,6 @@
                 android:padding="8dp"
                 />
 
-            <ImageButton
```
Comment (neslihanturan) [inline app/src/main/res/layout/layout_contribution.xml]: Instead of removing this button, please just add a lisstle space between this button and cancel button.

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java [hunk 5]
```diff
@@ -21,6 +21,7 @@
     @BindView(R.id.contributionSequenceNumber) TextView seqNumView;
     @BindView(R.id.contributionProgress) ProgressBar progressView;
     @BindView(R.id.failed_image_options) LinearLayout failedImageOptions;
+    @BindView(R.id.failed_image_options_retry) LinearLayout failedImageOptionsretry;
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java]: Ideally this PR should only edit .xml file. Since buttons are already defined and works fine. Please solve this issue by just editing .xml file, ie. adding a space between them

File: app/build.gradle (hunk 2 above)
Comment (sashrikakaur) [reply app/build.gradle]: I was just trying to test things around. You are right, these were unnecessary commits I'll remove them.

File: app/src/main/res/layout/layout_contribution.xml (hunk 4 above)
Comment (sashrikakaur) [reply app/src/main/res/layout/layout_contribution.xml]: I thought the retry button will provide a better UI, on the left side of the title. Should I remove that entirely,  keep them on the right side and only add some space between the buttons?

Comment (nicolas-raoul) [reply app/src/main/res/layout/layout_contribution.xml]: Yes just a bit more space between the two would be the best, I think :-)

File: app/build.gradle [hunk 6]
```diff
@@ -95,6 +95,7 @@ dependencies {
     implementation 'com.daimajia.swipelayout:library:1.2.0@aar'
     //metadata extractor
     implementation 'com.drewnoakes:metadata-extractor:2.11.0'
+
 }
```
Comment (neslihanturan) [inline app/build.gradle]: Please revert, not related with this PR

File: app/src/main/res/layout/layout_contribution.xml [hunk 7]
```diff
@@ -74,30 +74,31 @@
...
             android:orientation="horizontal"
-            android:layout_weight="1"
             android:visibility="gone"
             android:padding="@dimen/small_gap"
             >
-
             <ImageButton
-                android:id="@+id/cancelButton"
+                android:id="@+id/retryButton"
                 android:layout_width="wrap_content"
```
Comment (neslihanturan) [inline app/src/main/res/layout/layout_contribution.xml]: this PR just swaps cancel button with retry button. This is not what is needed on issue.

**Comment by neslihanturan** [comment]
Can you please also added updated screenshot after your code?

**Comment by sashrikakaur** [comment]
> Can you please also added updated screenshot after your code?

**Comment by sashrikakaur** [comment]
I have updated the screenshot. Sorry for the delay. Please have a look.
